File: MyAngular20.AppWebAngular/ClientApp/src/app/models/usuario-view.ts
export class UsuarioView {
  id: number = 0;
  nomeCompleto: string = "";
  usuarioNome: string = "";
  email: string = "";
  perfil: string = "";
  notificacoes: number = 0;
  senha: string = "";
  confirmaSenha: string = "";
  dtUltimoLogin: string = "";
  ativo: boolean = true;
  ativoStr: string = "";

  dataCadastro: string = "";
  cadastradoPor: string = "";
  dataAtualizacao: string = "";
  atualizadoPor: string = "";

  constructor() { }

  validar(): void {

    let regrasQuebradas: string = "";
    this.atualizadoPor = sessionStorage["nomeUsuario"];

    if (this.nomeCompleto === "")
      regrasQuebradas += "O campo Nome Completo é obrigatório.;";
    if (this.usuarioNome === "")
      regrasQuebradas += "O campo Nome Usuário é obrigatório.;";
    if (this.email === "")
      regrasQuebradas += "O campo Email é obrigatório.;";
    if (this.perfil === "")
      regrasQuebradas += "O campo Perfil é obrigatório.;";
    if (this.atualizadoPor === "")
      regrasQuebradas += "O campo Atualizado Por é obrigatório.;";
    if (this.senha !== this.confirmaSenha)
      regrasQuebradas != "Senha e Confirma Senha não conferem.;";

    if (this.id === 0) {
      if (this.senha === "")
        regrasQuebradas += "O campo Senha é obrigatório.;";
      if (this.confirmaSenha === "")
        regrasQuebradas += "O campo Confirma Senha é obrigatório.;";
    }

   

    if (regrasQuebradas.length > 0)
      throw new Error(regrasQuebradas);
    

  }


}

File: MyAngular20.ApplicationService/Views/UsuarioView.cs
namespace MyAngular20.ApplicationService.Views
{
    public class UsuarioView : ViewBase
    {
        public string NomeCompleto { get; set; }
        public string UsuarioNome { get; set; }
        public string Email { get; set; }
        public bool Ativo { get; set; }
        public string AtivoStr { get; set; }
        public string Perfil { get; set; }
        public string Senha { get; set; }
        public string ConfirmaSenha { get; set; }
        public int Notificacoes { get; set; }
        public string DataUltimoLogin { get; set; }
      


    }

    public interface IUsuarioFacade : IViewFacade<UsuarioView>
    {
        UsuarioView ObterPor(string nomeUsuario);

    }
}
